frontend/src/ClassView/ClassView.tsx

```diff
@@ -16,22 +18,86 @@ function ClassView() {
   const nav = useNavigate();
   const mode: "student" | "faculty" = "student"; // should be taken from some app state / login info
   const [cookies, setCookies, removeCookies] = useCookies(["jwt"]);
+  const [error, setError] = useState("");
+  const params = useParams();
+  const [classID, setClassID] = useState("");
+  const [assignments, setAssignments] = useState(new Array<any>());
+  getClassInfo(cookies.jwt);
```

Comment:
This will kill the performance of this page. You fire the function `getClassInfo` when the component draws. However, if you set the state of the component, it will redraw. This is an infinite loop which will crash the app.

Wrap it in an [`useEffect()`](https://reactjs.org/docs/hooks-reference.html#useeffect) call, then store the state of the response so you can use it for future navigation, rather than making a call every time.

```suggestion
  useEffect(() => {
    getClassInfo(cookies.jwt);
    // TODO: set the state of the component.
  }, [cookies.jwt]);
```